### PR TITLE
feat: new rule selector

### PR DESCRIPTION
**Pull request by stephanwampouille**
I believe I got everything in besides the actual math for the details

Screenshot 2023-02-07 at 4 19 17 PM

### CODE REVIEW

File: apps/detect/src/components/root.module.css [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
   display: flex;
   height: 100%;
   flex-direction: column;
   padding: 1.5rem 1rem;
 }
+
+.main {
+  display: flex;
+  flex: 1;
+  display: row;
```
Comment (ceci21) [inline apps/detect/src/components/root.module.css:19]: Do you mean `flex-direction: row`?

Comment (stephanwampouille) [reply apps/detect/src/components/root.module.css:19]: yes! good catch, thanks!

File: apps/detect/src/components/rule-selector/rule-selector.tsx [hunk 2]
```diff
@@ -1,62 +1,157 @@
...
 
 const RuleSelector = () => {
   const [expanded, setExpanded] = useState(false);
-  // const [expandedRule, setExpandedRule] = useState<Rule>();
+  const [expandedRule, setExpandedRule] = useState<Rule>();
+  const [ruleFilter, setRuleFilter] = useState("");
+
+  const rules = useQuery({
+    ...queries.detect.rules(),
+    suspense: false,
```
Comment (kenrick) [inline apps/detect/src/components/rule-selector/rule-selector.tsx:32]: This is can be `suspense: true`

Comment (kenrick) [reply apps/detect/src/components/rule-selector/rule-selector.tsx:32]: This way it loads the rules before showing the page

**Review by kenrick — CHANGES_REQUESTED** [review]
Looks great! small changes then good to job

**Review by kenrick — APPROVED** [review]

**Pull request merged**